### PR TITLE
Cancelling chart/release resource when chart had been cordon

### PR DESCRIPTION
**Pull request by tomahawk28**
Toward giantswarm/giantswarm#6078

Do the same practice as https://github.com/giantswarm/chart-operator/pull/250, but applied into `chart/release` resource.

### CODE REVIEW

File: service/controller/chart/v1/key/key.go [hunk 1]
```diff
@@ -31,6 +33,14 @@ func ConfigMapNamespace(customResource v1alpha1.Chart) string {
 	return customResource.Spec.Config.ConfigMap.Namespace
 }
 
+func CordonReason(customObject v1alpha1.Chart) string {
```
Comment (rossf7) [inline service/controller/chart/v1/key/key.go]:
```suggestion
func CordonReason(customResource v1alpha1.Chart) string {
```

This is the new stuff that will last a long time. So we should use the latest naming everywhere.

File: service/controller/chart/v1/resource/release/current.go [hunk 2]
```diff
@@ -2,11 +2,13 @@ package release
 
 import (
 	"context"
+	"fmt"
 
 	"github.com/giantswarm/helmclient"
 	"github.com/giantswarm/microerror"
 
 	"github.com/giantswarm/chart-operator/service/controller/chart/v1/key"
+	"github.com/giantswarm/operatorkit/controller/context/resourcecanceledcontext"
```
Comment (rossf7) [inline service/controller/chart/v1/resource/release/current.go]: Wrong block. Lets make the new stuff shiny 😉

File: service/controller/chart/v1/resource/release/current.go [hunk 3]
```diff
@@ -15,6 +17,15 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
...
 	}
 
+	if key.IsCordoned(cr) {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("release %#q has been cordoned until %#q due to reason %#q ", key.ReleaseName(cr), key.CordonUntil(cr), key.CordonReason(cr)))
+
+		resourcecanceledcontext.SetCanceled(ctx)
+		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+
+		return nil, nil
+	}
```
Comment (rossf7) [inline service/controller/chart/v1/resource/release/current.go]: This is all we need for this PR. But just a reminder the status resources for both controllers should set in the CR status that it is cordoned.

File: service/controller/chart/v1/resource/release/current.go (hunk 2 above)
Comment (tomahawk28) [reply service/controller/chart/v1/resource/release/current.go]: 😱, I thought that I saw all this, sorry.

File: service/controller/chart/v1/resource/release/current.go (hunk 3 above)
Comment (tomahawk28) [reply service/controller/chart/v1/resource/release/current.go]: Will bring it as next PR for chart-operator

File: service/controller/chart/v1/resource/release/current.go (hunk 2 above)
Comment (rossf7) [reply service/controller/chart/v1/resource/release/current.go]: Np ❤️ I forget this too sometimes!

File: service/controller/chart/v1/key/key.go [hunk 4]
```diff
@@ -47,6 +57,18 @@ func HasForceUpgradeAnnotation(customResource v1alpha1.Chart) bool {
 	return result
 }
 
+func IsCordoned(customResource v1alpha1.Chart) bool {
+	_, reasonOk := customResource.Annotations[annotation.CordonReason]
+	_, untilOk := customResource.Annotations[annotation.CordonUntilDate]
+
+	if reasonOk && untilOk {
```
Comment (kopiczko) [inline service/controller/chart/v1/key/key.go]: Same as in other PR. We should check the date. It should also be reflected in tests.

**Pull request merged**